app/models/metric/helper.rb

```diff
@@ -132,11 +132,11 @@ def self.remove_duplicate_timestamps(recs)
         _log.warn("Multiple rows found for the same timestamp: [#{rec.timestamp}], ids: [#{rec.id}, #{last_rec.id}], resource and id: [#{rec.resource_type}:#{rec.resource_id}]")
 
         # Merge records with the same timestamp
-        last_rec.attributes.each { |a| last_rec[a] ||= rec[a] }
+        last_rec.attribute_names.each { |a| last_rec[a] ||= rec[a] }
       else
         ret << rec
+        last_rec = rec
```

Comment:
It's not clear how the updated version of `last_rec` gets added to the `ret` array. But, it seems that, somehow, it is because there is a test case for it and it passes. Perhaps this can be rewritten to make easier to follow.